src/cr/cube/crunch_cube.py

```diff
@@ -200,82 +200,189 @@ def _as_array(self, include_missing=False, get_non_selected=False,
         )
         res = res + adjusted
 
+        # if prune and self.ndim < 3:
         if prune:
             return self._prune_body(res, include_transforms_for_dims)
 
         return res
 
+    def _prune_3d_body(self, res, transforms):
+        mask = np.zeros(res.shape)
+        for i, prune_inds in enumerate(self.prune_indices(transforms)):
+            rows_pruned = prune_inds[0]
+            cols_pruned = prune_inds[1]
+            rows_pruned = np.repeat(
+                rows_pruned[:, None], len(cols_pruned), axis=1
+            )
+            cols_pruned = np.repeat(
+                cols_pruned[None, :], len(rows_pruned), axis=0
+            )
+            slice_mask = np.logical_or(rows_pruned, cols_pruned)
+
+            if self.mr_dim_ind == (1, 2):
+                # In case of MR wee need to "inflate" mask
+                slice_mask = slice_mask[:, np.newaxis, :, np.newaxis]
```

Comment:
Does this work for both one MR and two MR situations?

Reply:
_Probably_ another good catch. Again, an example would be most welcome. Probably even necessary. Or at least a :ticket:

Reply:
It looks like we've got some cube examples here already:

mr x mr: https://github.com/Crunch-io/crunch-cube/blob/master/tests/integration/fixtures/cubes/mr-by-mr-different-mrs.json

cat x mr: https://github.com/Crunch-io/crunch-cube/blob/master/tests/integration/fixtures/cubes/selected-crosstab-array-last.json